### PR TITLE
WFLY-14436 Improve error for incorrect class for xa-datasource-class, etc.

### PR DESCRIPTION
**Pull request by sudeshnas93**
Issue: https://issues.redhat.com/browse/WFLY-14436

### CODE REVIEW

File: connector/src/main/java/org/jboss/as/connector/subsystems/datasources/JdbcDriverAdd.java [hunk 1]
```diff
@@ -115,18 +115,22 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
...
         if (xaDataSourceClassName != null) {
             Class<? extends XADataSource> dsCls;
             try {
                 dsCls = module.getClassLoader().loadClass(xaDataSourceClassName).asSubclass(XADataSource.class);
-            } catch (ClassNotFoundException | ClassCastException e) {
+            } catch (ClassNotFoundException e) {
                 throw SUBSYSTEM_DATASOURCES_LOGGER.failedToLoadDataSourceClass(xaDataSourceClassName, e);
-            }
+            } catch (ClassCastException e) {
+            throw SUBSYSTEM_DATASOURCES_LOGGER.notAValidDataSourceClass(dataSourceClassName, DataSource.class.getName());
```
Comment (kabir) [inline connector/src/main/java/org/jboss/as/connector/subsystems/datasources/JdbcDriverAdd.java:132]: This and the next line should be indented :)

Comment (sudeshnas93) [reply connector/src/main/java/org/jboss/as/connector/subsystems/datasources/JdbcDriverAdd.java]: Thanks @kabir . Done.

**Comment by kabir** [comment]
@tadamski please approve

**Review by tadamski — APPROVED** [review]
Approved

**Pull request merged**